src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py

```diff
@@ -1877,7 +1878,8 @@ def _get_public_ip_address_allocation(value, sku):
         vmss_dependencies.append('Microsoft.Network/applicationGateways/{}'.format(app_gateway))
 
         ag_dependencies = []
-        ag_dependencies.append('Microsoft.Network/virtualNetworks/{}'.format(vnet_name))
+        if app_gateway_type == 'new':
```

Comment:
this is redundant as the outer condition is already checking it

Reply:
also the condition should be ` if vnet_type == 'new':`

Reply:
Yes!  Thank you! My brain is mush!

Reply:
Ah, looks like you have already addressed it in a later commit

Reply:
No, I just did a commit --amend. It was all you.